src/hooks/useTermsAndConditions.ts

```diff
@@ -0,0 +1,38 @@
+import { useState } from 'preact/hooks';
+import { useLocalStorage } from './useLocalStorage';
+
+export const useTermsAndConditions = () => {
+  const { set, state } = useLocalStorage<string | undefined>({ key: 'TERMS_AND_CONDITIONS' });
+
+  // Terms and Conditions are accepted only when user has submitted the swap in `confirmSwap.ts`
+  const [termsAccepted, setTermsAccepted] = useState<boolean>(state === 'accepted');
+
+  // termsChecked is used to determine if the Terms and Conditions checkbox is checked and the Swap form can be submitted in `swap/index.tsx`
+  const [termsChecked, setTermsChecked] = useState<boolean>(false);
+
+  const toggleTermsChecked = () => {
+    setTermsChecked((state) => !state);
+  };
+
+  const acceptTerms = () => {
+    set('accepted');
+    setTermsAccepted(true);
+  };
+
+  const rejectTerms = () => {
+    set(undefined);
+    setTermsAccepted(false);
+  };
+
+  return {
+    termsChecked,
+    toggleTermsChecked,
+    termsAccepted,
+    acceptTerms,
+    rejectTerms,
+    setTermsAccepted: (accepted: boolean) => {
+      set(accepted ? 'accepted' : undefined);
+      setTermsAccepted(accepted);
+    },
```

Comment:
I think these three are somewhat redundant and we don't need to have all of them. Either we only expose `acceptTerms` and `rejectTerms` and remove `setTermsAccepted` or we only keep `setTermsAccepted.

I would suggest we do
```suggestion
    setTermsAccepted: (accepted: boolean) => {
      if (accepted) {
       acceptTerms()
      } else {
       rejectTerms()
      }
    },
```